diff -up ./qadic/ctx_init_conway.c.orig ./qadic/ctx_init_conway.c
--- ./qadic/ctx_init_conway.c.orig	2013-08-01 14:24:12.531737619 -0300
+++ ./qadic/ctx_init_conway.c	2013-08-01 14:26:18.526742444 -0300
@@ -25,6 +25,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <stdlib.h>
 
 #include "fmpz_vec.h"
 #include "padic.h"
@@ -53,8 +54,14 @@ void qadic_ctx_init_conway(qadic_ctx_t c
 
     if (!file)
     {
-        printf("Exception (qadic_ctx_init_conway).  File loading.\n");
-        abort();
+        char *path = getenv("FLINT_CPIMPORT");
+        if (path)
+            file = fopen(path, "r");
+        if (!file)
+        {
+            printf("Exception (qadic_ctx_init_conway).  File loading.\n");
+            abort();
+        }
     }
 
     while (fgets(buf, 832, file))
