diff -p -up flint-1.5.0/makefile.orig flint-1.5.0/makefile
--- flint-1.5.0/makefile.orig	2009-11-25 19:30:58.548075707 -0200
+++ flint-1.5.0/makefile	2009-11-25 19:32:14.061074816 -0200
@@ -345,56 +345,56 @@ NTL-interface-test.o: NTL-interface-test
 
 ####### test program targets
 
-mpn_extras-test: mpn_extras-test.o test-support.o $(FLINTOBJ) $(HEADERS)
-	$(CC) $(CFLAGS) mpn_extras-test.o test-support.o -o mpn_extras-test $(FLINTOBJ) $(LIBS)
+mpn_extras-test: mpn_extras-test.o test-support.o $(FLINT_LIB) $(HEADERS)
+	$(CC) $(CFLAGS) mpn_extras-test.o test-support.o -o mpn_extras-test $(FLINT_LIB) $(LIBS)
 
-fmpz_poly-test: fmpz_poly-test.o test-support.o $(FLINTOBJ) $(HEADERS)
-	$(CC) $(CFLAGS) fmpz_poly-test.o test-support.o -o fmpz_poly-test $(FLINTOBJ) $(LIBS)
+fmpz_poly-test: fmpz_poly-test.o test-support.o $(FLINT_LIB) $(HEADERS)
+	$(CC) $(CFLAGS) fmpz_poly-test.o test-support.o -o fmpz_poly-test $(FLINT_LIB) $(LIBS)
 
-fmpz-test: fmpz-test.o test-support.o $(FLINTOBJ) $(HEADERS)
-	$(CC) $(CFLAGS) fmpz-test.o test-support.o -o fmpz-test $(FLINTOBJ) $(LIBS)
+fmpz-test: fmpz-test.o test-support.o $(FLINT_LIB) $(HEADERS)
+	$(CC) $(CFLAGS) fmpz-test.o test-support.o -o fmpz-test $(FLINT_LIB) $(LIBS)
 
-F_mpz-test: F_mpz-test.o test-support.o $(FLINTOBJ) $(HEADERS)
-	$(CC) $(CFLAGS) F_mpz-test.o test-support.o -o F_mpz-test $(FLINTOBJ) $(LIBS)
+F_mpz-test: F_mpz-test.o test-support.o $(FLINT_LIB) $(HEADERS)
+	$(CC) $(CFLAGS) F_mpz-test.o test-support.o -o F_mpz-test $(FLINT_LIB) $(LIBS)
 
-ZmodF-test: ZmodF-test.o test-support.o $(FLINTOBJ) $(HEADERS)
-	$(CC) $(CFLAGS) ZmodF-test.o test-support.o -o ZmodF-test $(FLINTOBJ) $(LIBS)
+ZmodF-test: ZmodF-test.o test-support.o $(FLINT_LIB) $(HEADERS)
+	$(CC) $(CFLAGS) ZmodF-test.o test-support.o -o ZmodF-test $(FLINT_LIB) $(LIBS)
 
-ZmodF_poly-test: ZmodF_poly-test.o test-support.o $(FLINTOBJ) $(HEADERS)
-	$(CC) $(CFLAGS) ZmodF_poly-test.o test-support.o -o ZmodF_poly-test $(FLINTOBJ) $(LIBS)
+ZmodF_poly-test: ZmodF_poly-test.o test-support.o $(FLINT_LIB) $(HEADERS)
+	$(CC) $(CFLAGS) ZmodF_poly-test.o test-support.o -o ZmodF_poly-test $(FLINT_LIB) $(LIBS)
 
-mpz_poly-test: mpz_poly-test.o test-support.o $(FLINTOBJ) $(HEADERS)
-	$(CC) $(CFLAGS) mpz_poly-test.o test-support.o -o mpz_poly-test $(FLINTOBJ) $(LIBS)
+mpz_poly-test: mpz_poly-test.o test-support.o $(FLINT_LIB) $(HEADERS)
+	$(CC) $(CFLAGS) mpz_poly-test.o test-support.o -o mpz_poly-test $(FLINT_LIB) $(LIBS)
 
-F_mpz_mat-test: F_mpz_mat-test.o test-support.o $(FLINTOBJ) $(HEADERS)
-	$(CC) $(CFLAGS) F_mpz_mat-test.o test-support.o -o F_mpz_mat-test $(FLINTOBJ) $(LIBS)
+F_mpz_mat-test: F_mpz_mat-test.o test-support.o $(FLINT_LIB) $(HEADERS)
+	$(CC) $(CFLAGS) F_mpz_mat-test.o test-support.o -o F_mpz_mat-test $(FLINT_LIB) $(LIBS)
 
-F_mpz_LLL_fast_d-test: F_mpz_LLL_fast_d-test.o test-support.o $(FLINTOBJ) $(HEADERS)
-	$(CC) $(CFLAGS) F_mpz_LLL_fast_d-test.o test-support.o -o F_mpz_LLL_fast_d-test $(FLINTOBJ) $(LIBS)
+F_mpz_LLL_fast_d-test: F_mpz_LLL_fast_d-test.o test-support.o $(FLINT_LIB) $(HEADERS)
+	$(CC) $(CFLAGS) F_mpz_LLL_fast_d-test.o test-support.o -o F_mpz_LLL_fast_d-test $(FLINT_LIB) $(LIBS)
 	
-ZmodF_mul-test: ZmodF_mul-test.o test-support.o $(FLINTOBJ) $(HEADERS)
-	$(CC) $(CFLAGS) ZmodF_mul-test.o test-support.o -o ZmodF_mul-test $(FLINTOBJ) $(LIBS)
+ZmodF_mul-test: ZmodF_mul-test.o test-support.o $(FLINT_LIB) $(HEADERS)
+	$(CC) $(CFLAGS) ZmodF_mul-test.o test-support.o -o ZmodF_mul-test $(FLINT_LIB) $(LIBS)
 
-long_extras-test: long_extras-test.o test-support.o $(FLINTOBJ)
-	$(CC) $(CFLAGS) long_extras-test.o test-support.o -o long_extras-test $(FLINTOBJ) $(LIBS)
+long_extras-test: long_extras-test.o test-support.o $(FLINT_LIB)
+	$(CC) $(CFLAGS) long_extras-test.o test-support.o -o long_extras-test $(FLINT_LIB) $(LIBS)
 
-packed_vec-test: packed_vec-test.o test-support.o $(FLINTOBJ) $(HEADERS) 
-	$(CC) $(CFLAGS) packed_vec-test.o test-support.o -o packed_vec-test $(FLINTOBJ) $(LIBS)
+packed_vec-test: packed_vec-test.o test-support.o $(FLINT_LIB) $(HEADERS) 
+	$(CC) $(CFLAGS) packed_vec-test.o test-support.o -o packed_vec-test $(FLINT_LIB) $(LIBS)
 
-zmod_poly-test: zmod_poly-test.o test-support.o $(FLINTOBJ) $(HEADERS)
-	$(CC) $(CFLAGS) zmod_poly-test.o test-support.o -o zmod_poly-test $(FLINTOBJ) $(LIBS)
+zmod_poly-test: zmod_poly-test.o test-support.o $(FLINT_LIB) $(HEADERS)
+	$(CC) $(CFLAGS) zmod_poly-test.o test-support.o -o zmod_poly-test $(FLINT_LIB) $(LIBS)
 
-zmod_mat-test: zmod_mat-test.o test-support.o $(FLINTOBJ) $(HEADERS)
-	$(CC) $(CFLAGS) zmod_mat-test.o test-support.o -o zmod_mat-test $(FLINTOBJ) $(LIBS)
+zmod_mat-test: zmod_mat-test.o test-support.o $(FLINT_LIB) $(HEADERS)
+	$(CC) $(CFLAGS) zmod_mat-test.o test-support.o -o zmod_mat-test $(FLINT_LIB) $(LIBS)
 
-F_zmod_mat-test: F_zmod_mat-test.o test-support.o $(FLINTOBJ) $(HEADERS)
-	$(CC) $(CFLAGS) F_zmod_mat-test.o test-support.o -o F_zmod_mat-test $(FLINTOBJ) $(LIBS)
+F_zmod_mat-test: F_zmod_mat-test.o test-support.o $(FLINT_LIB) $(HEADERS)
+	$(CC) $(CFLAGS) F_zmod_mat-test.o test-support.o -o F_zmod_mat-test $(FLINT_LIB) $(LIBS)
 
-F_mpz_poly-test: F_mpz_poly-test.o test-support.o $(FLINTOBJ) $(HEADERS)
-	$(CC) $(CFLAGS) F_mpz_poly-test.o test-support.o -o F_mpz_poly-test $(FLINTOBJ) $(LIBS)
+F_mpz_poly-test: F_mpz_poly-test.o test-support.o $(FLINT_LIB) $(HEADERS)
+	$(CC) $(CFLAGS) F_mpz_poly-test.o test-support.o -o F_mpz_poly-test $(FLINT_LIB) $(LIBS)
 
-NTL-interface-test: NTL-interface.o NTL-interface-test.o test-support.o $(FLINTOBJ) $(HEADERS)
-	$(CPP) $(CFLAGS) NTL-interface-test.o NTL-interface.o test-support.o $(FLINTOBJ) -o NTL-interface-test $(LIBS2)
+NTL-interface-test: NTL-interface.o NTL-interface-test.o test-support.o $(FLINT_LIB) $(HEADERS)
+	$(CPP) $(CFLAGS) NTL-interface-test.o NTL-interface.o test-support.o $(FLINT_LIB) -o NTL-interface-test $(LIBS2)
 
 ####### tuning program object files
 
@@ -408,11 +408,11 @@ mpz_poly-tune.o: mpz_poly-tune.c $(HEADE
 
 ####### tuning program targets
 
-ZmodF_mul-tune: ZmodF_mul-tune.o test-support.o profiler.o $(FLINTOBJ) $(HEADERS)
-	$(CC) $(CFLAGS) ZmodF_mul-tune.o test-support.o profiler.o -o ZmodF_mul-tune $(FLINTOBJ) $(LIBS)
+ZmodF_mul-tune: ZmodF_mul-tune.o test-support.o profiler.o $(FLINT_LIB) $(HEADERS)
+	$(CC) $(CFLAGS) ZmodF_mul-tune.o test-support.o profiler.o -o ZmodF_mul-tune $(FLINT_LIB) $(LIBS)
 
-mpz_poly-tune: mpz_poly-tune.o test-support.o profiler.o $(FLINTOBJ) $(HEADERS)
-	$(CC) $(CFLAGS) mpz_poly-tune.o test-support.o profiler.o -o mpz_poly-tune $(FLINTOBJ) $(LIBS)
+mpz_poly-tune: mpz_poly-tune.o test-support.o profiler.o $(FLINT_LIB) $(HEADERS)
+	$(CC) $(CFLAGS) mpz_poly-tune.o test-support.o profiler.o -o mpz_poly-tune $(FLINT_LIB) $(LIBS)
 
 
 ####### profiling object files
@@ -487,7 +487,7 @@ bernoulli-profile.o: bernoulli-profile.c
 
 ####### profiling program targets
 
-PROFOBJ = $(FLINTOBJ) profiler.o profiler-main.o
+PROFOBJ = $(FLINT_LIB) profiler.o profiler-main.o
 
 fmpz_poly-profile: fmpz_poly-profile.o fmpz_poly-profile-tables.o test-support.o $(PROFOBJ)
 	$(CC) $(CFLAGS) -o fmpz_poly-profile fmpz_poly-profile.o fmpz_poly-profile-tables.o test-support.o $(PROFOBJ) $(LIBS)
@@ -505,8 +505,8 @@ ZmodF_poly-profile: ZmodF_poly-profile.o
 	$(CC) $(CFLAGS) -o ZmodF_poly-profile ZmodF_poly-profile.o ZmodF_poly-profile-tables.o $(PROFOBJ) $(LIBS)
 
 
-kara-profile: kara-profile.c profiler.o test-support.o $(FLINTOBJ)
-	$(CC) $(CFLAGS) -o kara-profile kara-profile.c profiler.o test-support.o $(FLINTOBJ) $(LIBS)
+kara-profile: kara-profile.c profiler.o test-support.o $(FLINT_LIB)
+	$(CC) $(CFLAGS) -o kara-profile kara-profile.c profiler.o test-support.o $(FLINT_LIB) $(LIBS)
 
 NTL-profile: NTL-profile.c test-support.o NTL-profile-tables.o $(PROFOBJ)
 	$(CPP) $(CFLAGS) -o NTL-profile NTL-profile.c NTL-profile-tables.o test-support.o $(PROFOBJ) $(LIB) -lntl
@@ -522,38 +522,38 @@ bernoulli-profile: bernoulli-profile.o b
 delta_qexp.o: delta_qexp.c $(HEADERS)
 	$(CC) $(CFLAGS) -c delta_qexp.c -o delta_qexp.o
 
-delta_qexp: delta_qexp.o $(FLINTOBJ)
-	$(CC) $(CFLAGS) -o delta_qexp delta_qexp.o $(FLINTOBJ) $(LIBS)
+delta_qexp: delta_qexp.o $(FLINT_LIB)
+	$(CC) $(CFLAGS) -o delta_qexp delta_qexp.o $(FLINT_LIB) $(LIBS)
 
-expmod: expmod.c $(FLINTOBJ)
-	$(CC) $(CFLAGS) -o expmod expmod.c $(FLINTOBJ) $(LIBS)
+expmod: expmod.c $(FLINT_LIB)
+	$(CC) $(CFLAGS) -o expmod expmod.c $(FLINT_LIB) $(LIBS)
 
-BPTJCubes: BPTJCubes.c $(FLINTOBJ)
-	$(CC) $(CFLAGS) -o BPTJCubes BPTJCubes.c $(FLINTOBJ) $(LIBS)
+BPTJCubes: BPTJCubes.c $(FLINT_LIB)
+	$(CC) $(CFLAGS) -o BPTJCubes BPTJCubes.c $(FLINT_LIB) $(LIBS)
 
 bernoulli.o: bernoulli.c $(HEADERS)
 	$(CC) $(CFLAGS) -c bernoulli.c -o bernoulli.o
 
-bernoulli: bernoulli.o $(FLINTOBJ)
-	$(CC) $(CFLAGS) -o bernoulli bernoulli.o $(FLINTOBJ) $(LIBS)
+bernoulli: bernoulli.o $(FLINT_LIB)
+	$(CC) $(CFLAGS) -o bernoulli bernoulli.o $(FLINT_LIB) $(LIBS)
 
 bernoulli_fmpz.o: bernoulli_fmpz.c $(HEADERS)
 	$(CC) $(CFLAGS) -c bernoulli_fmpz.c -o bernoulli_fmpz.o
 
-bernoulli_fmpz: bernoulli_fmpz.o $(FLINTOBJ)
-	$(CC) $(CFLAGS) -o bernoulli_fmpz bernoulli_fmpz.o $(FLINTOBJ) $(LIBS)
+bernoulli_fmpz: bernoulli_fmpz.o $(FLINT_LIB)
+	$(CC) $(CFLAGS) -o bernoulli_fmpz bernoulli_fmpz.o $(FLINT_LIB) $(LIBS)
 
 bernoulli_zmod.o: bernoulli_zmod.c $(HEADERS)
 	$(CC) $(CFLAGS) -c bernoulli_zmod.c -o bernoulli_zmod.o
 
-bernoulli_zmod: bernoulli_zmod.o $(FLINTOBJ)
-	$(CC) $(CFLAGS) -o bernoulli_zmod bernoulli_zmod.o $(FLINTOBJ) $(LIBS)
+bernoulli_zmod: bernoulli_zmod.o $(FLINT_LIB)
+	$(CC) $(CFLAGS) -o bernoulli_zmod bernoulli_zmod.o $(FLINT_LIB) $(LIBS)
 
 thetaproduct.o: thetaproduct.c $(HEADERS)
 	$(CC) $(CFLAGS2) $(WITH_ZNPOLY) -c thetaproduct.c -o thetaproduct.o
 
-thetaproduct: thetaproduct.o $(FLINTOBJ)
-	$(CC) $(CFLAGS2) -o thetaproduct thetaproduct.o $(FLINTOBJ) $(LIBS)
+thetaproduct: thetaproduct.o $(FLINT_LIB)
+	$(CC) $(CFLAGS2) -o thetaproduct thetaproduct.o $(FLINT_LIB) $(LIBS)
 
 ####### Quadratic sieve
 
@@ -590,13 +590,13 @@ mp_lprels.o: QS/mp_lprels.c QS/mp_lprels
 mp_factor_base.o: QS/mp_factor_base.c QS/mp_factor_base.h
 	$(CC) $(CFLAGS) -c QS/mp_factor_base.c -o mp_factor_base.o
 
-mpQS: QS/mpQS.c QS/mpQS.h QS/tinyQS.h mp_factor_base.o mp_poly.o mp_sieve.o mp_linear_algebra.o mp_lprels.o $(FLINTOBJ)
-	$(CC) $(CFLAGS) -o mpQS QS/mpQS.c mp_factor_base.o mp_poly.o mp_sieve.o mp_linear_algebra.o mp_lprels.o $(FLINTOBJ) $(LIBS)
+mpQS: QS/mpQS.c QS/mpQS.h QS/tinyQS.h mp_factor_base.o mp_poly.o mp_sieve.o mp_linear_algebra.o mp_lprels.o $(FLINT_LIB)
+	$(CC) $(CFLAGS) -o mpQS QS/mpQS.c mp_factor_base.o mp_poly.o mp_sieve.o mp_linear_algebra.o mp_lprels.o $(FLINT_LIB) $(LIBS)
 
 ####### Integer multiplication timing
 
 ZMULOBJ = zn_mod.o misc.o mul_ks.o pack.o mul.o mulmid.o mulmid_ks.o ks_support.o mpn_mulmid.o nuss.o pmf.o pmfvec_fft.o tuning.o mul_fft.o mul_fft_dft.o array.o invert.o zmod_mat.o zmod_poly.o memory-manager.o fmpz.o ZmodF_mul-tuning.o mpz_poly.o mpz_poly-tuning.o fmpz_poly.o ZmodF_poly.o mpz_extras.o profiler.o ZmodF_mul.o ZmodF.o mpn_extras.o F_mpz_mul-timing.o long_extras.o factor_base.o poly.o sieve.o linear_algebra.o block_lanczos.o
 
-F_mpz_mul-timing: $(FLINTOBJ) 
-	$(CC) $(CFLAGS) F_mpz_mul-timing.c profiler.o -o Zmul $(FLINTOBJ) $(LIBS)
+F_mpz_mul-timing: $(FLINT_LIB) 
+	$(CC) $(CFLAGS) F_mpz_mul-timing.c profiler.o -o Zmul $(FLINT_LIB) $(LIBS)
 
