diff -p -up flint-1.5.0/makefile.orig flint-1.5.0/makefile
--- flint-1.5.0/makefile.orig	2009-11-25 19:24:12.705078707 -0200
+++ flint-1.5.0/makefile	2009-11-25 19:24:37.694075743 -0200
@@ -148,7 +148,8 @@ libflint.dll: $(FLINTOBJ)
 	$(CC) -fPIC -shared -o libflint.dll $(FLINTOBJ) $(LIBS)
 
 libflint.so: $(FLINTOBJ)
-	$(CC) -fPIC -shared -o libflint.so $(FLINTOBJ) $(LIBS)
+	$(CC) -fPIC -shared -Wl,-soname,libflint.so.$(FLINT_VERSION) -o libflint.so.$(FLINT_VERSION) $(FLINTOBJ) $(LIBS)
+	ln -s libflint.so.$(FLINT_VERSION) libflint.so
 
 ##### zn_poly object files
 
