diff -up ./configure.orig ./configure
--- configure.orig	2014-01-02 12:08:34.000000000 -0700
+++ configure	2014-02-12 20:00:00.000000000 -0700
@@ -111,6 +111,9 @@ while [ "$1" != "" ]; do
          FLINT_CPIMPORT_DIR="$PREFIX/share/flint"
          FLINT_CPIMPORT="$FLINT_CPIMPORT_DIR/CPimport.txt"
          ;;
+      --libdir)
+         LIBDIR=$VALUE
+         ;;
       --enable-shared)
          SHARED=1
          ;;
@@ -171,6 +174,12 @@ while [ "$1" != "" ]; do
       CFLAGS)
          CFLAGS="$VALUE"
          ;;
+      CXXFLAGS)
+         CXXFLAGS="$VALUE"
+         ;;
+      LDFLAGS)
+         LDFLAGS="$VALUE"
+         ;;
       ABI)
          ABI="$VALUE"
          ;;
@@ -186,15 +195,12 @@ done
 
 LIBS="m"
 
-if [ -d "${GMP_DIR}/lib" ]; then
-   GMP_LIB_DIR="${GMP_DIR}/lib"
-   GMP_INC_DIR="${GMP_DIR}/include"
-elif [ -d "${GMP_DIR}/lib64" ]; then
-   GMP_LIB_DIR="${GMP_DIR}/lib64"
-   GMP_INC_DIR="${GMP_DIR}/include"
-elif [ -d "${GMP_DIR}/.libs" ]; then
+if [ -d "${GMP_DIR}/.libs" ]; then
    GMP_LIB_DIR="${GMP_DIR}/.libs"
    GMP_INC_DIR="${GMP_DIR}"
+elif [ -d "${GMP_DIR}" ]; then
+   GMP_LIB_DIR="${GMP_DIR}"
+   GMP_INC_DIR="${GMP_DIR}/../include"
 else
    echo "Invalid GMP directory"
    exit 1
@@ -203,18 +209,12 @@ LIB_DIRS="${LIB_DIRS} ${GMP_LIB_DIR}"
 INC_DIRS="${INC_DIRS} ${GMP_INC_DIR}"
 LIBS="${LIBS} gmp"
 
-if [ -d "${MPFR_DIR}/lib" ]; then
-   MPFR_LIB_DIR="${MPFR_DIR}/lib"
-   MPFR_INC_DIR="${MPFR_DIR}/include"
-elif [ -d "${MPFR_DIR}/lib64" ]; then
-   MPFR_LIB_DIR="${MPFR_DIR}/lib64"
-   MPFR_INC_DIR="${MPFR_DIR}/include"
-elif [ -d "${MPFR_DIR}/.libs" ]; then
-   MPFR_LIB_DIR="${MPFR_DIR}/.libs"
-   MPFR_INC_DIR="${MPFR_DIR}"
-elif [ -d "${MPFR_DIR}/src/.libs" ]; then
+if [ -d "${MPFR_DIR}/src/.libs" ]; then
    MPFR_LIB_DIR="${MPFR_DIR}/src/.libs"
    MPFR_INC_DIR="${MPFR_DIR}/src"
+elif [ -d "${MPFR_DIR}" ]; then
+   MPFR_LIB_DIR="${MPFR_DIR}"
+   MPFR_INC_DIR="${MPFR_DIR}/../include"
 else
    echo "Invalid MPFR directory"
    exit 1
@@ -226,12 +226,9 @@ LIBS="${LIBS} mpfr"
 #configure extra libraries
 
 if [ "$WANT_NTL" = "1" ]; then
-   if [ -d "${NTL_DIR}/lib" ]; then
-      NTL_LIB_DIR="${NTL_DIR}/lib"
-      NTL_INC_DIR="${NTL_DIR}/include"
-   elif [ -d "${NTL_DIR}/lib64" ]; then
-      NTL_LIB_DIR="${NTL_DIR}/lib64"
-      NTL_INC_DIR="${NTL_DIR}/include"
+   if [ -d "${NTL_DIR}" ]; then
+      NTL_LIB_DIR="${NTL_DIR}"
+      NTL_INC_DIR="${NTL_DIR}/../include"
    else
       echo "Invalid NTL directory"
       exit 1
@@ -616,6 +613,7 @@ echo "FLINT_SHARED=$SHARED" >> Makefile
 echo "FLINT_LIB=$FLINT_LIB" >> Makefile
 echo "EXEEXT=$EXEEXT" >> Makefile
 echo "PREFIX=$PREFIX" >> Makefile
+echo "LIBDIR=$LIBDIR" >> Makefile
 echo "" >> Makefile
 echo "WANT_NTL=$WANT_NTL" >> Makefile
 echo "" >> Makefile
@@ -631,6 +629,8 @@ echo "CXX=$CXX" >> Makefile
 echo "AR=$AR" >> Makefile
 echo "" >> Makefile
 echo "CFLAGS=$CFLAGS" >> Makefile
+echo "CXXFLAGS=$CXXFLAGS" >> Makefile
+echo "LDFLAGS=$LDFLAGS" >> Makefile
 echo "ABI_FLAG=$ABI_FLAG" >> Makefile
 echo "PIC_FLAG=$PIC_FLAG" >> Makefile
 echo "" >> Makefile
diff -up ./Makefile.in.orig ./Makefile.in
--- Makefile.in.orig	2014-01-02 12:08:34.000000000 -0700
+++ Makefile.in	2014-02-12 20:00:00.000000000 -0700
@@ -102,10 +102,10 @@
 	$(AT)$(foreach dir, $(BUILD_DIRS), mkdir -p build/$(dir); BUILD_DIR=../build/$(dir); export BUILD_DIR; MOD_DIR=$(dir); export MOD_DIR; $(MAKE) -f ../Makefile.subdirs -C $(dir) shared || exit $$?;)
 	$(AT)if [ "$(WANT_NTL)" -eq "1" ]; then \
 	  $(MAKE) build/interfaces/NTL-interface.lo; \
-	  $(CXX) $(ABI_FLAG) -shared build/interfaces/NTL-interface.lo $(LOBJS) $(MOD_LOBJS) $(EXT_OBJS) $(LIBS2) -o $(FLINT_LIB); \
+	  $(CXX) $(ABI_FLAG) -shared build/interfaces/NTL-interface.lo $(LDFLAGS) $(LOBJS) $(MOD_LOBJS) $(LIBS2) -o $(FLINT_LIB) -Wl,-h,$(FLINT_LIB); \
 	fi
 	$(AT)if [ "$(WANT_NTL)" -ne "1" ]; then \
-	  $(CC) $(ABI_FLAG) -shared $(LOBJS) $(MOD_LOBJS) $(EXT_OBJS) $(LIBS2) -o $(FLINT_LIB); \
+	  $(CC) $(ABI_FLAG) -shared $(LDFLAGS) $(LOBJS) $(MOD_LOBJS) $(LIBS2) -o $(FLINT_LIB) -Wl,-h,$(FLINT_LIB); \
 	fi
 
 libflint.a: $(OBJS) $(LIB_SOURCES) $(EXT_SOURCES) $(HEADERS) $(EXT_HEADERS) | build build/interfaces
@@ -169,23 +169,26 @@
 endif
 
 install: library
-	mkdir -p $(DESTDIR)$(PREFIX)/lib
+	mkdir -p $(DESTDIR)$(LIBDIR)
 	mkdir -p $(DESTDIR)$(PREFIX)/include/flint
 	$(AT)if [ "$(FLINT_SHARED)" -eq "1" ]; then \
-		cp $(FLINT_LIB) $(DESTDIR)$(PREFIX)/lib; \
+		cp -p $(FLINT_LIB) $(DESTDIR)$(LIBDIR); \
 	fi
 	$(AT)if [ "$(FLINT_STATIC)" -eq "1" ]; then \
-		cp libflint.a $(DESTDIR)$(PREFIX)/lib; \
+		cp -p libflint.a $(DESTDIR)$(LIBDIR); \
 	fi
 	cp $(HEADERS) $(DESTDIR)$(PREFIX)/include/flint
 	$(AT)if [ ! -z $(EXT_HEADERS) ]; then \
-		cp $(EXT_HEADERS) $(DESTDIR)$(PREFIX)/include/flint; \
+		cp -p $(EXT_HEADERS) $(DESTDIR)$(PREFIX)/include/flint; \
+	fi
+	$(AT)if [ "$(WANT_NTL)" -eq "1" ]; then \
+		cp -p NTL-interface.h $(DESTDIR)$(PREFIX)/include/flint; \
 	fi
 	mkdir -p $(DESTDIR)$(FLINT_CPIMPORT_DIR)
-	cp qadic/CPimport.txt $(DESTDIR)$(FLINT_CPIMPORT_DIR)
+	cp -p qadic/CPimport.txt $(DESTDIR)$(FLINT_CPIMPORT_DIR)
 	mkdir -p $(DESTDIR)$(PREFIX)/include/flint/flintxx
-	cp flintxx/*.h $(DESTDIR)$(PREFIX)/include/flint/flintxx
-	cp *xx.h $(DESTDIR)$(PREFIX)/include/flint
+	cp -p flintxx/*.h $(DESTDIR)$(PREFIX)/include/flint/flintxx
+	cp -p *xx.h $(DESTDIR)$(PREFIX)/include/flint
 
 build:
 	mkdir -p build
@@ -206,13 +209,13 @@
 	mkdir -p build/interfaces
 
 build/interfaces/NTL-interface.lo: interfaces/NTL-interface.cpp NTL-interface.h
-	$(QUIET_CXX) $(CXX) $(PIC_FLAG) $(CFLAGS) $(INCS) -c $< -o $@;
+	$(QUIET_CXX) $(CXX) $(PIC_FLAG) $(CXXFLAGS) $(INCS) -c $< -o $@;
 
 build/interfaces/NTL-interface.o: interfaces/NTL-interface.cpp NTL-interface.h
-	$(QUIET_CXX) $(CXX) $(CFLAGS) $(INCS) -c $< -o $@;
+	$(QUIET_CXX) $(CXX) $(CXXFLAGS) $(INCS) -c $< -o $@;
 
 build/interfaces/test/t-NTL-interface$(EXEEXT): interfaces/test/t-NTL-interface.cpp
-	$(QUIET_CXX) $(CXX) $(CFLAGS) $(INCS) $< build/interfaces/NTL-interface.o -o $@ $(LIBS);
+	$(QUIET_CXX) $(CXX) $(CXXFLAGS) $(INCS) $< build/interfaces/NTL-interface.o -o $@ $(LIBS);
 
 print-%:
 	@echo '$*=$($*)'
